astropy/coordinates/earth.py

```diff
@@ -174,6 +179,73 @@ def from_geodetic(cls, lon, lat, height=0., ellipsoid=None):
         self._ellipsoid = ellipsoid
         return self.to(height.unit)
 
+    @classmethod
+    def of_site(cls, site_name):
+        """
+        Return an object of this class for a known observatory/site by name.
+
+        Note that this is intended as a quick convenience function, not a
+        fully-featured registry of observatories.
+
+        Parameters
+        ----------
+        site_name : str
+            Name of the observatory.
+
+        Returns
+        -------
+        site : This class (a `~astropy.coordinates.EarthLocation` or subclass)
+            The location of the observatory.
+
+        See Also
+        --------
+        get_site_names : the list of sites that this function can access
+        """
+        # need to import inside function to avoid circular dependencies
+        from .sites import get_site
+
+        try:
+            el = get_site(site_name, online=True)  # this is always an EarthLocation
+        except URLError:
```

Comment:
Hmm, I *thought* (but could be wrong) that `URLError` is the base error to try to catch here.  `urlopen` says it throws things that are  `URLError` or subclasses, anyway. I guess I could just catch *everything*... Do you think that's better, @embray?